pkg/ddl/ddl_worker.go

```diff
@@ -434,15 +462,62 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
 			logutil.BgLogger().Info("pause user DDL by system successful", zap.String("category", "ddl-upgrading"), zap.Stringer("job", job))
 		}
 
+		if job.Version == model.TiDBDDLV2 {
+			toLocalWorker = true
+			if _, err := job.Encode(true); err != nil {
+				return err
+			}
+		}
+
 		jobTasks = append(jobTasks, job)
 		injectModifyJobArgFailPoint(job)
 	}
 
 	se.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
 
+	if toLocalWorker {
```

Comment:
Could we remove `toLocalWorker`? We can use `job.Version == model.TiDBDDLV2` directly.